Plugins are not loaded if the entry is not on the same directory as package.json

<!---
Thanks for filing an issue 😄 ! Before you submit, please read the following:

Search open/closed issues before submitting since someone might have asked the same thing before!
-->

This is a 🐛 bug report.

<!--- Provide a general summary of the issue in the title above -->

### 🎛 Directory structure
<!--- If describing a bug, tell us what your babel configuration looks like -->

```
- package.json
+ src
  - main.js
+ node_modules
  + parcel-bundler
  + parcel-plugin-plugin
```

### 🤔 Expected Behavior
Parcel should be able to load the plugins from the `package.json`.
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

### 😯 Current Behavior
It does not if the entry file is not in the root directory of the project.

The error is silenced but if we log it we get :
```
Error: Cannot find module './package.json' from '[project]/src'
at Function.module.exports [as sync] ([project]/node_modules/resolve/lib/sync.js:40:15)
at module.exports ([project]/node_modules/parcel-bundler/src/utils/localRequire.js:11:24)
at Bundler.loadPlugins ([project]/node_modules/parcel-bundler/src/Bundler.js:86:17)
at new Bundler ([project]/node_modules/parcel-bundler/src/Bundler.js:44:10)
at Command.bundle ([project]/node_modules/parcel-bundler/bin/cli.js:66:19)
at Command.listener ([project]/node_modules/commander/index.js:315:8)
at emitTwo (events.js:126:13)
at Command.emit (events.js:214:7)
at Command.parseArgs ([project]/node_modules/commander/index.js:653:12)
at Command.parse ([project]/node_modules/commander/index.js:475:21) code: 'MODULE_NOT_FOUND'
```

<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If you are seeing an error, please include the full error message and stack trace -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

### 💁 Possible Solution

I dirty-fixed it by replacing `./package.json` to `../package.json`, but the best solution should be to traverse the directory tree until a `package.json` is found.
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

### 🔦 Context

Found this while working on a plugin
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

### 🌍 Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

| Software         | Version(s)
| ---------------- | ----------
| Parcel           |  1.1.0
| Node             | 8.9.1
| npm/Yarn         | 1.3.2
| Operating System | macOS 10.13.2

--- a/src/Bundler.js
+++ b/src/Bundler.js
@@ -87,16 +87,21 @@ class Bundler extends EventEmitter {
     this.packagers.add(type, packager);
   }
 
-  loadPlugins() {
+  loadPlugins(location = this.mainFile) {
     try {
-      let pkg = localRequire('./package.json', this.mainFile);
+      let pkg = localRequire('./package.json', location);
       let deps = Object.assign({}, pkg.dependencies, pkg.devDependencies);
       for (let dep in deps) {
         if (dep.startsWith('parcel-plugin-')) {
-          localRequire(dep, this.mainFile)(this);
+          localRequire(dep, location)(this);
         }
       }
     } catch (err) {
+      let currPath = Path.dirname(this.mainFile);
+      currPath = currPath.substring(0, currPath.lastIndexOf('/'));
+      if (currPath.length > 1) {
+        return this.loadPlugins(currPath);
+      }
       this.logger.warn(err);
     }
   }

--- a/src/Bundler.js
+++ b/src/Bundler.js
@@ -88,8 +88,15 @@ class Bundler extends EventEmitter {
   }
 
   loadPlugins(location = this.mainFile) {
+    let pkg;
     try {
-      let pkg = localRequire('./package.json', location);
+      pkg = localRequire('./package.json', location);
+    } catch (err) {
+      if (err.code === 'MODULE_NOT_FOUND' && location.length > 1) {
+        return this.loadPlugins(Path.join(location, '..'));
+      }
+    }
+    try {
       let deps = Object.assign({}, pkg.dependencies, pkg.devDependencies);
       for (let dep in deps) {
         if (dep.startsWith('parcel-plugin-')) {
@@ -97,9 +104,6 @@ class Bundler extends EventEmitter {
         }
       }
     } catch (err) {
-      if (err.code === 'MODULE_NOT_FOUND' && location.length > 1) {
-        return this.loadPlugins(Path.join(location, '..'));
-      }
       this.logger.warn(err);
     }
   }

--- a/src/Bundler.js
+++ b/src/Bundler.js
@@ -97,10 +97,8 @@ class Bundler extends EventEmitter {
         }
       }
     } catch (err) {
-      let currPath = Path.dirname(this.mainFile);
-      currPath = currPath.substring(0, currPath.lastIndexOf('/'));
-      if (currPath.length > 1) {
-        return this.loadPlugins(currPath);
+      if (err.code === 'MODULE_NOT_FOUND' && location.length > 1) {
+        return this.loadPlugins(Path.join(location, '..'));
       }
       this.logger.warn(err);
     }

--- a/test/integration/plugins/sub-folder/index.js
+++ b/test/integration/plugins/sub-folder/index.js
@@ -1,0 +1,1 @@
+module.exports = require('./test.txt');

--- a/test/integration/plugins/sub-folder/test.txt
+++ b/test/integration/plugins/sub-folder/test.txt
@@ -1,0 +1,1 @@
+hello world

--- a/test/plugins.js
+++ b/test/plugins.js
@@ -14,4 +14,17 @@ describe('plugins', function() {
     let output = run(b);
     assert.equal(output, 'hello world');
   });
+
+  it('should load package.json from parent tree', async function() {
+    let b = await bundle(__dirname + '/integration/plugins/sub-folder/index.js');
+
+    assertBundleTree(b, {
+      name: 'index.js',
+      assets: ['index.js', 'test.txt'],
+      childBundles: []
+    });
+
+    let output = run(b);
+    assert.equal(output, 'hello world');
+  });
 });

